consensus/hotstuff/votecollector/combined_vote_processor_v2.go

```diff
@@ -0,0 +1,232 @@
+package votecollector
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/rs/zerolog"
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/consensus/hotstuff/signature"
+	"github.com/onflow/flow-go/consensus/hotstuff/verification"
+	"github.com/onflow/flow-go/model/encoding"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	msig "github.com/onflow/flow-go/module/signature"
+)
+
+/* **************** Base-Factory for CombinedVoteProcessors ***************** */
+
+// combinedVoteProcessorFactoryBaseV2 is a `votecollector.baseFactory` for creating
+// CombinedVoteProcessors, holding all needed dependencies.
+// combinedVoteProcessorFactoryBaseV2 is intended to be used for the main consensus.
+// CAUTION:
+// this base factory only creates the VerifyingVoteProcessor for the given block.
+// It does _not_ check the proposer's vote for its own block, i.e. it does _not_
+// implement `hotstuff.VoteProcessorFactory`. This base factory should be wrapped
+// by `votecollector.VoteProcessorFactory` which adds the logic to verify
+// the proposer's vote (decorator pattern).
+type combinedVoteProcessorFactoryBaseV2 struct {
```

Comment:
Not sure we need to name it `V2`, because we don't have `V1`.

I would suggest to just remove it.

Reply:
But we have v3, I would say if we have v3 then v2 makes sense